### PR TITLE
Add feature for Micronaut AWS APIGateway (dependency and docs links)

### LINKED ISSUE #1549: Add feature for Micronaut AWS APIGateway
see: https://github.com/micronaut-projects/micronaut-aws/pull/1517

### PR DESCRIPTION
**Pull request by wetted**
closes #1549

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java [hunk 1]
```diff
@@ -178,13 +178,20 @@ private void populateDependencies(GeneratorContext generatorContext) {
                 .artifactId("junit-jupiter-engine")
                 .test());
         if (generatorContext.getFeatures().hasFeature(AmazonApiGatewayHttp.class)) {
+            dependencyContext.addDependency(MicronautDependencyUtils.awsDependency()
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java]: we don't need this

File: starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java [hunk 2]
```diff
@@ -178,13 +178,20 @@ private void populateDependencies(GeneratorContext generatorContext) {
...
+                    .compile());
             dependencyContext.addDependency(Dependency.builder()
                     .lookupArtifactId("apigatewayv2-alpha")
                     .compile());
             dependencyContext.addDependency(Dependency.builder()
                     .lookupArtifactId("apigatewayv2-integrations-alpha")
                     .compile());
-
+        }
+        if (generatorContext.getFeatures().hasFeature(AmazonApiGateway.class)) {
```
Comment (sdelamo) [inline starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java]: we don't need this

File: starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy [hunk 3]
```diff
@@ -35,4 +38,27 @@ class AmazonApiGatewaySpec extends ApplicationContextSpec implements CommandOutp
...
         amazonApiGateway.supports(ApplicationType.DEFAULT)
     }
+
+    void 'amazon-api-gateway feature with Cdk has dependency in infra and doc links'() {
+        when:
+        def output = generate(ApplicationType.FUNCTION, new Options(Language.JAVA, BuildTool.GRADLE),
+                [Cdk.NAME, AmazonApiGateway.NAME])
+
+        then:
+        output."$Cdk.INFRA_MODULE/build.gradle".contains($/implementation("io.micronaut.aws:micronaut-aws-apigateway/$)
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy]: we don't need this

File: starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy [hunk 4]
```diff
@@ -35,4 +38,27 @@ class AmazonApiGatewaySpec extends ApplicationContextSpec implements CommandOutp
...
+        output."README.md".contains($/https://docs.aws.amazon.com/apigateway/$)
+    }
+
+    void 'amazon-api-gateway feature without Cdk has dependency in project and doc links'() {
+        when:
+        def output = generate(ApplicationType.FUNCTION, new Options(Language.JAVA, BuildTool.GRADLE),
+                [AmazonApiGateway.NAME])
+
+        then:
+        output."build.gradle".contains($/implementation("io.micronaut.aws:micronaut-aws-apigateway/$)
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy]: we don't need this

**Review by sdelamo — CHANGES_REQUESTED** [review]
Can you create a OneOfFeature for AmzonAPIGateway. Both `AmazonApiGateway` and `AmazonApiGatewayHttp` should extend this new class.

**Comment by wetted** [comment]
> Can you create a OneOfFeature for AmzonAPIGateway. Both `AmazonApiGateway` and `AmazonApiGatewayHttp` should extend this new class.

It already is and there is a test for that in `AmazonApiGatewayHttpSpec`

File: starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java (hunk 1 above)
Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java]: removed

File: starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java (hunk 2 above)
Comment (wetted) [reply starter-core/src/main/java/io/micronaut/starter/feature/aws/Cdk.java]: removed

File: starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy (hunk 4 above)
Comment (wetted) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy]: removed

File: starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy (hunk 3 above)
Comment (wetted) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy]: removed

File: starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy [hunk 5]
```diff
@@ -35,4 +38,35 @@ class AmazonApiGatewaySpec extends ApplicationContextSpec implements CommandOutp
         amazonApiGateway.supports(ApplicationType.FUNCTION)
         amazonApiGateway.supports(ApplicationType.DEFAULT)
     }
+
+    void 'amazon-api-gateway feature with Cdk has dependency in infra and doc links'() {
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy]: test description is incorrect

File: starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy [hunk 6]
```diff
@@ -35,4 +38,35 @@ class AmazonApiGatewaySpec extends ApplicationContextSpec implements CommandOutp
...
+        when:
+        def output = generate(ApplicationType.FUNCTION, new Options(Language.JAVA, BuildTool.GRADLE),
+                [Cdk.NAME, AmazonApiGateway.NAME])
+
+        then:
+        output."README.md".contains($/https://micronaut-projects.github.io/micronaut-aws/latest/guide/index.html#amazonApiGateway/$)
+        output."README.md".contains($/https://docs.aws.amazon.com/apigateway/$)
+    }
+
+    void 'amazon-api-gateway feature without Cdk has dependency in project and doc links'() {
```
Comment (sdelamo) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/aws/AmazonApiGatewaySpec.groovy]: test description is incorrect

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**